drivers/cc2420/cc2420.c

```diff
@@ -216,32 +217,43 @@ void cc2420_tx_exec(cc2420_t *dev)
 int cc2420_rx(cc2420_t *dev, uint8_t *buf, size_t max_len, void *info)
 {
     uint8_t len;
+    uint8_t crc_corr;
 
-    /* get the packet length (without dropping it) (first byte in RX FIFO */
+    /* get the packet length (without dropping it) (first byte in RX FIFO) */
     cc2420_ram_read(dev, CC2420_RAM_RXFIFO, &len, 1);
+
+
     len -= 2;   /* subtract RSSI and FCF */
 
     if (!buf) {
         DEBUG("cc2420: recv: packet of length %i in RX FIFO\n", (int)len);
+        return 0;
```

Comment:
Please remove `return 0` here and set `len=0`. Additionally, use `else` for the code below which should be executed when `buf` is actually set